scripts/prepare_release.sh

```diff
@@ -55,7 +55,7 @@ for GITIGNORE in $GITIGNORES; do
         sed -i 's/###START_COMMENTED_GENERATED_FILES###/###START_GENERATED_FILES###/' $GITIGNORE
         sed -i 's/###END_COMMENTED_GENERATED_FILES###/###END_GENERATED_FILES###/' $GITIGNORE
     else
-        sed -i '/###START_GENERATED_FILES###/,/###END_GENERATED_FILES###/s/^/# /' $GITIGNORE
+        sed -i '/###START_GENERATED_FILES###/,/###END_GENERATED_FILES###/s/^/#/' $GITIGNORE
```

Comment:
This isn't the correct fix.

The original line was putting a comment + space at the beginning of each line from `START_GENERATED_FILES` to `END_GENERATED_FILES` ready for the `sed` at line 54 to remove later in "unrelease" mode.

By removing this space, line 54 will no longer remove the comment, so this script will be broken

Reply:
I would argue that this half the fix. As @tom-cosgrove-arm said the unrelease expression in line 54 needs to be updated as well.

`sed -i '/###START_COMMENTED_GENERATED_FILES###/,/###END_COMMENTED_GENERATED_FILES###/s/^#//' $GITIGNORE`

@mcagriaksoy  could you please update the PR and test it using:

~~~~~
# Prepare a release
./scripts/prepare_release.sh &&  cat programs/.gitignore

# Check that there is no trailing whitespace added
 tests/scripts/check_files.py

# Return to devel mode and make sure the git status is clean
./scripts/prepare_release.sh -u
~~~~~~~~~~

Thanks for your contribution.

Reply:
As I've mentioned before, I think we should remove `-u` mode. We don't actually need it. Our process for “return to devel mode” is `git revert`, not running a script.